src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.Streaming.cs

```diff
@@ -157,6 +159,85 @@ await foreach (var result in results.WithCancellation(combinedState.Cancellation
                     pagingContinuation);
             }
 
+            public class DocumentLastModifiedComparer : Comparer<ShardStreamItem<BlittableJsonReaderObject>>
+            {
+                public override int Compare(ShardStreamItem<BlittableJsonReaderObject> x, ShardStreamItem<BlittableJsonReaderObject> y)
+                {
+                    if (x?.Item == null)
+                        return -1;
+                    if (y?.Item == null)
+                        return -1;
+
+                    if (x.Item.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject md1) && md1.TryGet(Constants.Documents.Metadata.LastModified, out DateTime lm1) &&
```

Comment:
I think we have TryGetMetadata and TryGetLastModified extension methods?

Reply:
Yep. Changed